api/db/database-builder/factory/build-badge.js

```diff
@@ -1,18 +1,20 @@
 import { databaseBuffer } from '../database-buffer.js';
 import { buildTargetProfile } from './build-target-profile.js';
 import _ from 'lodash';
+import { randomUUID } from 'crypto';
 
 function buildBadge({
   id = databaseBuffer.getNextId(),
   altMessage = 'alt message',
   imageUrl = '/img_funny.svg',
   message = 'message',
   title = 'title',
-  key = 'key',
+  key,
   isCertifiable = false,
   targetProfileId,
   isAlwaysVisible = false,
 } = {}) {
+  key = _.isUndefined(key) ? `key_${randomUUID()}` : key;
```

Comment:
Question qui m'a été posée : ce changement est lié au fait qu'il existe une contrainte de BDD d'unicité sur ce champ. Pour pas casser tous les tests qui commit souvent un seul badge dans les tests (j'ai des tests qui en commit plusieurs), ou bien qui donnent le nom, j'ai ajouté cela pour être à la fois rétrocompatible, et pour ne pas avoir à reseigner cela dans mes given alors que c'est un champ dont je me moque totalement.

TLDR : tout comme `id` qui est dynamique car doit être unique et not null, key à une contrainte d'unicité not null et il n'est donc pas logique pas défaut d'être fixe à 'key'